plugins/wasi_nn/ggml.cpp

```diff
@@ -459,6 +460,93 @@ ErrNo EvaluateTokens(Graph &GraphRef, struct llama_context *LlamaContext,
   return ErrNo::Success;
 }
 
+const std::string_view Base64ImageTagPrefix = "<img src=\"data:image/";
+const std::string_view Base64ImageBytesPrefix = ";base64,";
+const std::string_view Base64ImageTagSuffix = "\">";
+const std::string_view PromptImagePlaceholder = "<image>";
+
+bool checkBase64Image(const std::string Prompt) noexcept {
+  // Check if the prompt contains a base64 image.
+  // Follow this link for the supported image formats:
+  // https://github.com/ggerganov/llama.cpp/blob/master/common/stb_image.h
+
+  auto Base64ImageTagBeginPos = Prompt.find(Base64ImageTagPrefix);
+  if (Base64ImageTagBeginPos == std::string::npos) {
+    return false;
+  }
+  auto Base64ImageTagEndPos =
+      Prompt.find(Base64ImageTagSuffix, Base64ImageTagBeginPos);
```

Comment:
Do we want to have debug logging here to tell users that the base64 is broken here?